src/toil/test/jobStores/jobStoreTest.py

```diff
@@ -1333,39 +1322,31 @@ def _prepareTestFile(self, bucket, size=None):
             return url
         read_type = 'r' if USING_PYTHON2 else 'rb'
         with open('/dev/urandom', read_type) as readable:
-            if USING_PYTHON2:
-                bucket.new_key(fileName).set_contents_from_string(readable.read(size))
-            else:
-                bucket.new_key(fileName).set_contents_from_string(str(readable.read(size)))
-        return url, hashlib.md5(bucket.get_key(fileName).get_contents_as_string()).hexdigest()
+            bucket.put_object(Key=fileName, Body=str(readable.read(size)))
+        return url, hashlib.md5(bucket.Object(fileName).get()['Body'].read()).hexdigest()
 
     def _hashTestFile(self, url):
         from toil.jobStores.aws.jobStore import AWSJobStore
-        key = AWSJobStore._getKeyForUrl(urlparse.urlparse(url), existing=True)
-        try:
-            contents = key.get_contents_as_string()
-        finally:
-            key.bucket.connection.close()
+        key = AWSJobStore._getObjectForUrl(urlparse.urlparse(url), existing=True)
+        contents = key.get()['Body'].read()
         return hashlib.md5(contents).hexdigest()
 
     def _createExternalStore(self):
-        import boto.s3
+        """A S3.Bucket instance is returned"""
         from toil.jobStores.aws.utils import region_to_bucket_location
-        s3 = boto.s3.connect_to_region(self.awsRegion())
-        try:
-            return s3.create_bucket(bucket_name='import-export-test-%s' % uuid.uuid4(),
-                                    location=region_to_bucket_location(self.awsRegion()))
-        except:
-            with panic(log=logger):
-                s3.close()
+        from toil.jobStores.aws.jobStore import get_boto3_session
+        resource = get_boto3_session().resource('s3', region_name=self.awsRegion())
+        bucket = resource.Bucket('import-export-test-%s' % uuid.uuid4())
+        bucket.create(CreateBucketConfiguration={'LocationConstraint': region_to_bucket_location(self.awsRegion())})
+        bucket.wait_until_exists()
+        return bucket
 
     def _cleanUpExternalStore(self, bucket):
         try:
-            for key in bucket.list():
-                key.delete()
+            bucket.objects.all().delete()
             bucket.delete()
-        finally:
-            bucket.connection.close()
+        except:
```

Comment:
You no longer need the `try` here and if bucket deletion fails, it should error.

Formerly this was a `try` `finally`, which only gaurantees that the `finally` clause runs in the event of an error, but stills errors.